x/superfluid/keeper/concentrated_liquidity_test.go

```diff
@@ -0,0 +1,247 @@
+	cltypes "github.com/osmosis-labs/osmosis/v15/x/concentrated-liquidity/types"
+	"github.com/osmosis-labs/osmosis/v15/x/superfluid/keeper"
+	"github.com/osmosis-labs/osmosis/v15/x/superfluid/types"
+)
+
+func (suite *KeeperTestSuite) TestAddToConcentratedLiquiditySuperfluidPosition() {
+	defaultJoinTime := suite.Ctx.BlockTime()
+	type sendTest struct {
+		superfluidDelegated    bool
+		superfluidUndelegating bool
+		unlocking              bool
+		overwritePositionId    bool
+		amount0Added           sdk.Int
+		amount1Added           sdk.Int
+		doNotFundAcc           bool
+		isLastPositionInPool   bool
+		expectedError          error
+	}
+	testCases := map[string]sendTest{
+		"add to position that is superfluid delegated, not unlocking": {
+			superfluidDelegated: true,
+			amount0Added:        sdk.NewInt(100000000),
+			amount1Added:        sdk.NewInt(100000000),
+		},
+		"add to position that is superfluid delegated, not unlocking, not enough funds to add": {
+			doNotFundAcc:        true,
+			superfluidDelegated: true,
+			amount0Added:        sdk.NewInt(100000000),
+			amount1Added:        sdk.NewInt(100000000),
+			expectedError:       fmt.Errorf("insufficient funds"),
+		},
+		"error: last position in pool": {
+			superfluidDelegated:  true,
+			isLastPositionInPool: true,
+			amount0Added:         sdk.NewInt(100000000),
+			amount1Added:         sdk.NewInt(100000000),
+			expectedError:        cltypes.AddToLastPositionInPoolError{PoolId: 1, PositionId: 1},
+		},
+		"error: lock that is not superfluid delegated, not unlocking": {
+			amount0Added:  sdk.NewInt(100000000),
+			amount1Added:  sdk.NewInt(100000000),
+			expectedError: types.ErrNotSuperfluidUsedLockup,
+		},
+		"error: lock that is not superfluid delegated, unlocking": {
+			unlocking:     true,
+			amount0Added:  sdk.NewInt(100000000),
+			amount1Added:  sdk.NewInt(100000000),
+			expectedError: types.LockImproperStateError{LockId: 1, UnbondingDuration: suite.App.StakingKeeper.UnbondingTime(suite.Ctx).String()},
+		},
+		"error: lock that is superfluid undelegating, not unlocking": {
+			superfluidDelegated:    true,
+			superfluidUndelegating: true,
+			amount0Added:           sdk.NewInt(100000000),
+			amount1Added:           sdk.NewInt(100000000),
+			expectedError:          types.ErrNotSuperfluidUsedLockup,
+		},
+		"error: lock that is superfluid undelegating, unlocking": {
+			superfluidDelegated:    true,
+			superfluidUndelegating: true,
+			unlocking:              true,
+			amount0Added:           sdk.NewInt(100000000),
+			amount1Added:           sdk.NewInt(100000000),
+			expectedError:          types.LockImproperStateError{LockId: 1, UnbondingDuration: suite.App.StakingKeeper.UnbondingTime(suite.Ctx).String()},
+		},
+		"error: non-existent position ID": {
+			overwritePositionId: true,
+			amount0Added:        sdk.NewInt(100000000),
+			amount1Added:        sdk.NewInt(100000000),
+			expectedError:       cltypes.PositionIdNotFoundError{PositionId: 5},
+		},
```

Comment:
Can we add test case for 
- Amount0Added negative
- Amount1Added negative?

Reply:
added neg amt0 and amt1 tests here https://github.com/osmosis-labs/osmosis/pull/5042/commits/8540cf1b30394a11e5b9fab2f17609cda60b29a8